File: src/dataset/go/resistor_color_duo_Exercism_go_thinkverse/openAI/ChatGPT4_resistor_color_duo_v4.go
package resistorcolorduo

var _colors = [10]string{"black", "brown", "red", "orange", "yellow", "green", "blue", "violet", "grey", "white"}

// Value should return the resistance value of a resistor with a given colors.
func Value(colors []string) int {
	if len(colors) < 2 {
		return 0
	}

	colorMap := func(color string) int {
		for i, c := range _colors {
			if c == color {
				return i
			}
		}
		return 0
	}

	return (colorMap(colors[0]) * 10) + colorMap(colors[1])
}

File: src/dataset/typescript/resistor-color-duo_Exercism-typescript-alemarr/openAI/ChatGPT4_resistor-color-duo_v4.ts
const colorsMap = Object.freeze({
  black: 0,
  brown: 1,
  red: 2,
  orange: 3,
  yellow: 4,
  green: 5,
  blue: 6,
  violet: 7,
  grey: 8,
  white: 9,
});

export const decodedValue = ([color1, color2]: [keyof typeof colorsMap, keyof typeof colorsMap]): number =>
  colorsMap[color1] * 10 + colorsMap[color2];